### PR TITLE
SW-5325 Updated To-Do items priority logic

### PR DESCRIPTION
**Pull request by tommylau523**
The following are the requirements:
1) 6 Items total
2) In To-do sections, 3 deliverables at max (to make room for events)

Screenshots:
1) Overdue item in To-Do, current event in to-do

2) 3 Overdue items, no other deliverables

3) 3 Overdue items, 2 events, 1 other deliverables, 1 deliverable cut off

### CODE REVIEW

**Comment by tommylau523** [comment]
* **#2616** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2616?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tommylau523 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by nickgraz — APPROVED** [review]
Nice work

File: src/scenes/Home/ParticipantHomeView/ToDoProvider/index.tsx [hunk 1]
```diff
@@ -49,36 +47,58 @@ const ToDoProvider = ({ children }: Props) => {
...
-    setToDoItems(allItems.filter((item) => item.getSection() == 'To Do').slice(0, 3));
-    setUpcomingItems(allItems.filter((item) => item.getSection() == 'Upcoming').slice(0, 3));
-  }, [allItems, setToDoItems, setUpcomingItems]);
+      const sortedDeliverables = allDeliverables.sort(compareToDoItems);
+      const sortedEvents = allEvents.sort(compareToDoItems);
+
+      const toDoDeliverables = sortedDeliverables.filter((deliverable) => deliverable.getSection() === 'To Do');
+      const upcomingDeliverables = sortedDeliverables.filter((deliverable) => deliverable.getSection() === 'Upcoming');
+
+      console.log(upcomingDeliverables);
```
[the remark this answers is not in the document]
Comment (tommylau523) [reply src/scenes/Home/ParticipantHomeView/ToDoProvider/index.tsx]: Removed. I checked and found that the tslint rules still has the console part. Will investigate further.

[the remark this answers is not in the document]
Comment (constanzauanini) [reply src/scenes/Home/ParticipantHomeView/ToDoProvider/index.tsx]: Maybe it stopped working when I updated from tslint to eslint (with the react version update)

**Pull request merged**